File: trunk/net.tenteCsharp/src_code/windowManagement/FloorGUI.java


cclass windowManagement.WindowManagement; 

public cclass FloorGUI extends TypeComponent{ 

	public BlindNotifyPort blindNotifyPort;
		public WindowNotifyPort windowNotifyPort;
	
	
	public FloorGUI (String id){
		super(id);
	}
	
	public BlindNotifyPort getBlindNotifyPort(){
		return blindNotifyPort;
	}
	
	public void setBlindNotifyPort(FloorGUI.BlindNotifyPort value){
		this.blindNotifyPort=value;
	}
	
	public WindowNotifyPort getWindowNotifyPort(){
		return windowNotifyPort;
	}
	
	public void setWindowNotifyPort(FloorGUI.WindowNotifyPort value){
		this.windowNotifyPort=value;
	}
	

	
	
	public cclass WindowNotifyPort extends TypePort{

		public ArrayList portsIGeneralWindowNotify;

		public WindowNotifyPort (){
			super();
			portsIGeneralWindowNotify=new ArrayList();

		}


			
		public void connectPort(IGeneralWindowNotify port){
			portsIGeneralWindowNotify.add(port);
		}
	}	
	
	public cclass BlindNotifyPort extends TypePort{

		public ArrayList portsIGeneralBlindNotify;

		public BlindNotifyPort (){
			super();
			portsIGeneralBlindNotify=new ArrayList();

		}


			
		public void connectPort(IGeneralBlindNotify port){
			portsIGeneralBlindNotify.add(port);
		}
	}

}

File: pseudoCodeGeneratorElio/src-gen/windowManagement/FloorGUI.cs
using System;
using System.Collections;
using System.Text;

namespace SmartHome
{
	public partial class FloorGUI : TypeComponent
	{
		public BlindNotifyPort blindNotifyPort;
			public WindowNotifyPort windowNotifyPort;
	
		public FloorGUI (String id) 
			: base(id)
		{
		}
		public BlindNotifyPort getBlindNotifyPort()
		{
			return blindNotifyPort;
		}
	
		public void setBlindNotifyPort(FloorGUI.BlindNotifyPort value)
		{
			this.blindNotifyPort=value;
		}
	
		public WindowNotifyPort getWindowNotifyPort()
		{
			return windowNotifyPort;
		}
	
		public void setWindowNotifyPort(FloorGUI.WindowNotifyPort value)
		{
			this.windowNotifyPort=value;
		}
	

	
		
		public class WindowNotifyPort : TypePort
		{
 		public ArrayList portsIGeneralWindowNotify = new ArrayList();

			public WindowNotifyPort()
				: base()
			{
				portsIGeneralWindowNotify=new ArrayList();

			}


			public ArrayList getPortsIGeneralWindowNotify()
			{
	        	return portsIGeneralWindowNotify;
	        }
				
			public void connectPort(IGeneralWindowNotify port)
			{
				portsIGeneralWindowNotify.Add(port);
			}
	
		}	
		
		public class BlindNotifyPort : TypePort
		{
 		public ArrayList portsIGeneralBlindNotify = new ArrayList();

			public BlindNotifyPort()
				: base()
			{
				portsIGeneralBlindNotify=new ArrayList();

			}


			public ArrayList getPortsIGeneralBlindNotify()
			{
	        	return portsIGeneralBlindNotify;
	        }
				
			public void connectPort(IGeneralBlindNotify port)
			{
				portsIGeneralBlindNotify.Add(port);
			}
	
		}
	}
}